### PR TITLE
Implement auto-approvals for datadog

### PR DESCRIPTION
**Pull request by bernardjkim**
Paired with https://github.com/gravitational/teleport.e/pull/5150

This PR extends the Datadog plugin to support automatic approvals.

Example Usage: A user with the following role will have access requests automatically approved if they are on-call for the `datadog-team` at the time of making the access request.
```yaml
kind: role
metadata:
  name: requester
spec:
  allow:
    request:
      annotations:
        teleport.dev/schedules:
        - datadog-team
...
```

changelog: Extend the Datadog plugin to support automatic approvals.

### CODE REVIEW

File: integrations/access/accessrequest/app.go
Comment (hugoShaka) [inline integrations/access/accessrequest/app.go:580]: I was going to complain about the fragile string-based error handling but we are doing this in every other plugin 😭

File: integrations/access/accessrequest/app.go
Comment (hugoShaka) [inline integrations/access/accessrequest/app.go:274]: I think we should run this only if `teleportUser` is not empty. This will reduce the noise against the Teleport cluster from non-configured plugins.

File: integrations/access/datadog/bot.go
Comment (hugoShaka) [inline integrations/access/datadog/bot.go:41]: With new plugins we tried to introduce new standard annotations (with very moderate success) following the pattern `teleport.dev/something`.

Existing annotations are:
- `teleport.dev/notify-services` to choose who we should notify (deprecated, AMRs are a better way to do this)
- `teleport.dev/schedules` to specify the on-call schedule names for auto-approval
- `teleport.dev/teams` to choose which team we should notify (yes this makes absolutely no sense, this should be `notify-teams`)

The concepts of schedules and teams might not make sense for every service provider, we can either:
- introduce another annotation (but the `teams` annotation is now coined 😫 )
- reuse an existing one (only the `teleport.dev/schedules` does auto-approval today)

Those constants are stored there: https://github.com/gravitational/teleport/blob/4b3fe19d972391152d56b1744efd4bcfa32956dd/api/types/constants.go#L792-L797

File: integrations/access/datadog/client.go
Comment (hugoShaka) [inline integrations/access/datadog/client.go:62]: IS there a link or something describing the API endpoint? Somewhere we can check if the API has been moved out unstable. I failed to find public documentation about this endpoint.

**Review by hugoShaka — COMMENTED** [review]
Looks good, thank you for adding the auto-approval capability into the common app.

File: integrations/access/datadog/client.go
Comment (bernardjkim) [reply integrations/access/datadog/client.go:62]: Not that I'm aware of. This API endpoint has only been shared with us via email communication. I think we'll just need to link to the general API references page and keep an eye out for when documentation becomes available.

File: integrations/access/datadog/bot.go
Comment (bernardjkim) [reply integrations/access/datadog/bot.go:41]: After some more thought, is there any concern with this annotation being reused for multiple plugins? I could see a user unintentionally adding automatic approvals for another plugin's service/team with the same names.

For example, if I have a Datadog team that has the same name as an PagerDuty service, then there is no option to enable automatic approvals for only the Datadog team.

File: integrations/access/accessrequest/app.go
Comment (bernardjkim) [reply integrations/access/accessrequest/app.go:274]: The `teleportUser` is not explicitly required for "self-hosted" plugins, and I don't think we should make it explicitly required since that will add more complexity to the configuration.

The function should log in debug level in most cases and only log a warning if the code execution reaches the point where it attempts to submit an access review.

File: integrations/access/datadog/bot.go
Comment (hugoShaka) [reply integrations/access/datadog/bot.go]: > After some more thought, is there any concern with this annotation being reused for multiple plugins? I could see a user unintentionally adding automatic approvals for another plugin's service/team with the same names.

I agree that this is a questionable design but this is what was done for the last plugins. We should get rid of annotations entirely by moving approval via AMRs as well. In the meantime I'd recommend to stick to the current broken plan rather than changing again. This is not a blocking comment, you might want to check with Roman before merging.

File: integrations/access/datadog/bot.go
Comment (hugoShaka) [inline integrations/access/datadog/bot.go]: Follow up on [the `teleportUser` discussion](https://github.com/gravitational/teleport/pull/47475/files/d4ce6a432625e29a48af6f31cdaf91c6091e5141#r1797467035): it seems this code will run on every request. In this case can we try to fail as fast as possible?

Here we could save many datadog calls by checking the annotations before looking up the on-call teams and returning early if auto-approval is not configured.

File: integrations/access/datadog/bot.go
Comment (bernardjkim) [reply integrations/access/datadog/bot.go]: That sounds reasonable. I'm okay with sticking to the current annotation pattern for now and improving the workflow with AMRs in the future.

File: integrations/access/datadog/bot.go
Comment (bernardjkim) [reply integrations/access/datadog/bot.go]: Good call

**Review by hugoShaka — APPROVED** [review]

**Review by marcoandredinis — APPROVED** [review]

**Pull request merged**
